### PR TITLE
[Static web assets] Reduce allocations on DefineStaticWebAssetEndpoints

### PR DESCRIPTION
**Pull request by javiercn**
* Moves the builtin mappings from an ItemGroup to a static table on the task to avoid repeatedly creating the items as well as the mappings.

### CODE REVIEW

File: src/StaticWebAssetsSdk/Tasks/Data/ContentTypeProvider.cs [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+    {
+        var extension = Path.GetExtension(relativePath);
+        if (extension == ".gz" || extension == ".br")
+        {
+            var fileName = Path.GetFileNameWithoutExtension(relativePath);
+            if (Path.GetExtension(fileName) != "")
+            {
+                var result = ResolveBuiltIn(fileName, log);
+                // If we don't have a specific mapping for the other extension, use any mapping available for `.gz` or `.br`
+                return result.Equals(default) && _builtInMappings.TryGetValue(extension, out var compressed) ?
```
Comment (MackinnonBuck) [inline src/StaticWebAssetsSdk/Tasks/Data/ContentTypeProvider.cs:436]: `ContentTypeMapping` doesn't override `Equals()`, so `result.Equals(default)` will always return `false`, right?

Comment (MackinnonBuck) [reply src/StaticWebAssetsSdk/Tasks/Data/ContentTypeProvider.cs:436]: Maybe `ResolveBuiltIn()` should return a `ContentTypeMapping?`.

File: src/StaticWebAssetsSdk/Tasks/Data/ContentTypeProvider.cs [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+            [".thn"] = new ContentTypeMapping("application/octet-stream", null, "*.thn", 1),
+            [".toc"] = new ContentTypeMapping("application/octet-stream", null, "*.toc", 1),
+            [".u32"] = new ContentTypeMapping("application/octet-stream", null, "*.u32", 1),
+            [".xsn"] = new ContentTypeMapping("application/octet-stream", null, "*.xsn", 1),
+            [".xtp"] = new ContentTypeMapping("application/octet-stream", null, "*.xtp", 1),
+        };
+
+    internal ContentTypeMapping ResolveContentTypeMapping(string relativePath, TaskLoggingHelper log)
+    {
+        var builtIn = ResolveBuiltIn(relativePath, log);
```
Comment (MackinnonBuck) [inline src/StaticWebAssetsSdk/Tasks/Data/ContentTypeProvider.cs:408]: It looks like `ResolveBuiltIn()` gets called regardless of whether a custom mapping is resolved. Is that necessary?

File: src/StaticWebAssetsSdk/Tasks/Data/ContentTypeProvider.cs (hunk 1 above)
Comment (javiercn) [reply src/StaticWebAssetsSdk/Tasks/Data/ContentTypeProvider.cs]: I've changed the comparison

**Review by MackinnonBuck — APPROVED** [review]

**Pull request merged**